public/javascript/pump/socket.js

```diff
@@ -149,24 +149,30 @@
             }
         };
 
-        sock.onclose = function(e) {
+        sock.onclose = function(error) {
             Pump.socket = null;
             // Reconnect on broken connections, not for normal close
-            if (e.code !== 1000) {
-                sock.reconnect(e);
+            if (error.code !== 1000) {
+                sock.reconnect(error);
             }
         };
 
-        sock.reconnect = function(e) {
+        sock.reconnect = function(error) {
             _socket.retryAttempts++;
 
             if (_socket.retryTimer) {
                 clearTimeout(_socket.retryTimer);
                 _socket.retryTimer = null;
             }
 
-            if (e.code === 2000 && _socket.retryAttempts > 10) {
+            if (error.code === 2000 && _socket.retryAttempts > 10) {
                 // No more attempts if all transports failed
+                if (!_socket.error) {
+                    // Show alert only one time
+                    _socket.error = error;
+                    Pump.error("Looks like your browser not support any socket transport");
```

Comment:
Does SockJS just call onclose immediately with no error if it's not supported? That seems kinda funky.

Also, I think this would be clearer if it were something like "Sorry! Your browser doesn't support realtime."

Reply:
Yes, but only try to connect to be sure.